pkg/controller/istiocontrolplane/customizer.go

```diff
@@ -29,13 +30,20 @@ var _ helmreconciler.RenderingCustomizerFactory
 
 // NewCustomizer returns a RenderingCustomizer for Istio
 func (f *IstioRenderingCustomizerFactory) NewCustomizer(instance runtime.Object) (helmreconciler.RenderingCustomizer, error) {
-	istioControlPlane, ok := instance.(*v1alpha1.IstioControlPlane)
-	if !ok {
+	switch v := instance.(type) {
+	case *v1alpha1.IstioControlPlane:
+		return &helmreconciler.SimpleRenderingCustomizer{
+			InputValue:          NewIstioRenderingInput(v),
+			PruningDetailsValue: NewIstioPruningDetails(v),
+			ListenerValue:       NewIstioRenderingListener(v),
+		}, nil
+	case *v1alpha2.IstioControlPlane:
+		return &helmreconciler.SimpleRenderingCustomizer{
+			InputValue:          NewIstioRenderingInputV2(v),
+			PruningDetailsValue: NewIstioPruningDetailsV2(v),
+			ListenerValue:       NewIstioRenderingListenerV2(v),
+		}, nil
```

Comment:
@ostromart @richardwxn , are we supporting both v1alph1 and v1alph2 at the same time?

Reply:
After discussing with @rcernich we are removing v1alpha1 as no-one will be using this going forward.

Reply:
ok i think that is reasonable. I'm quite confused when I saw v1alpha1 which is pretty much based on helm values if i understand it correctly.